imports/client/components/SetupPage.tsx

```diff
@@ -468,6 +469,488 @@ class GoogleIntegrationSection extends React.Component<GoogleIntegrationSectionP
+    };
+    const formOptions = [noneOption, ...this.props.guilds];
+    return (
+      <div>
+        {this.state.submitState === 'submitting' ? <Alert variant="info">Saving...</Alert> : null}
+        {this.state.submitState === 'success' ? <Alert variant="success" dismissible onClose={this.dismissAlert}>Saved changes.</Alert> : null}
+        {this.state.submitState === 'error' ? (
+          <Alert variant="danger" dismissible onClose={this.dismissAlert}>
+            Saving failed:
+            {' '}
+            {this.state.submitError}
+          </Alert>
+        ) : null}
+
+        <form onSubmit={this.onSaveGuild}>
+          <FormGroup>
+            <FormLabel htmlFor="jr-setup-edit-discord-bot-guild">
+              Bot token
+            </FormLabel>
+            <FormControl
+              id="jr-setup-edit-discord-bot-guild"
+              as="select"
+              type="text"
+              placeholder=""
+              value={this.state.guildId}
+              disabled={shouldDisableForm}
+              onChange={this.onSelectedGuildChange}
+            >
+              {formOptions.map(({ _id, name }) => {
+                return (
+                  <option key={_id} value={_id}>{name}</option>
+                );
+              })}
+            </FormControl>
+          </FormGroup>
+          <Button variant="primary" type="submit" onClick={this.onSaveGuild} disabled={shouldDisableForm}>Save</Button>
+        </form>
+      </div>
+    );
+  }
+}
+
+const DiscordGuildFormContainer = withTracker((_props: DiscordGuildFormContainerProps) => {
+  // DiscordGuilds is a pseudocollection, and the subscribe here causes the
+  // server to do a call against the Discord API to list guilds the user is in.
+  // It's not reactive; you might have to refresh the page to get it to update.
+  // Didn't seem worth making cleverer; this will get used ~once.
+  const guildSub = Meteor.subscribe('discord.guilds');
+  const ready = guildSub.ready();
+  const guilds = ready ? DiscordGuilds.find({}).fetch() : [];
+  return {
+    ready: false,
+    guilds,
+  };
+})(DiscordGuildForm);
+
+interface DiscordIntegrationSectionProps {
+  configured: boolean;
+  enabled: boolean;
+  oauthSettings?: Configuration;
+  botToken?: string;
+  guild?: DiscordGuildType;
+}
+
+class DiscordIntegrationSection extends React.Component<DiscordIntegrationSectionProps> {
+  onToggleEnabled = () => {
+    const newValue = !this.props.enabled;
+    const ffValue = newValue ? 'off' : 'on';
+    Meteor.call('setFeatureFlag', 'disable.discord', ffValue);
+  };
+
+  render() {
+    const firstButtonLabel = this.props.enabled ? 'Enabled' : 'Enable';
+    const secondButtonLabel = this.props.enabled ? 'Disable' : 'Disabled';
+
+    const configured = !!this.props.oauthSettings;
+    const clientId = this.props.oauthSettings && this.props.oauthSettings.appId;
+    const addGuildLink = `https://discord.com/api/oauth2/authorize?client_id=${clientId}&scope=bot&permissions=8`;
```

Comment:
FYI, `permissions=8` here is `ADMINISTRATOR`, which may be more than you intend to give out. I wasn't sure the universe of things that we'd want our bot to be able to do, so for my testing purposes I just asked for everything.  Also, I imagine that since you've already added the bot to the guild in prod, we won't need to click this link anyway.

Feel free to request that we replace this with a narrower permissions set for the guild owner to be granting to the application (bot).

Reply:
I'm pretty sure I used `ADMINISTRATOR` when I set it up. I, too, am not sure what all I want the bot to be able to do, but I don't really have any objection to it being able to do everything. (I suspect, for instance, that we'll end up wanting to play with things like permissions and so on).